### PR TITLE
Allow AWS Provider ACM Certificate Data Source to Return TLS Certificate Material

### PR DESCRIPTION
**Pull request by rizkybiz**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Relates OR Closes #0000

Output from acceptance testing:

```
$ make testacc TESTS=TestAccXXX PKG=ec2

...
```

### CODE REVIEW

**Review by anGie44 — CHANGES_REQUESTED** [review]
Hi @rizkybiz, thanks again for this PR! It's off to a great start, just a couple notes to address in addition to a request to update the test file (`internal/service/acm/certificate_data_source_test.go`) with the following lines to check for the new expected values. 

```go
func TestAccACMCertificateDataSource_singleIssued(t *testing.T) {
	...

	resource.ParallelTest(t, resource.TestCase{
		PreCheck:   func() { acctest.PreCheck(t) },
		ErrorCheck: acctest.ErrorCheck(t, acm.EndpointsID),
		Providers:  acctest.Providers,
		Steps: []resource.TestStep{
			{
				Config: testAccCheckCertificateDataSourceConfig(domain),
				Check: resource.ComposeTestCheckFunc(
					//lintignore:AWSAT001
					resource.TestMatchResourceAttr(resourceName, "arn", arnRe),
					resource.TestCheckResourceAttr(resourceName, "status", acm.CertificateStatusIssued),
					resource.TestCheckResourceAttrSet(resourceName, "certificate"),
					resource.TestCheckResourceAttrSet(resourceName, "certificate_chain"),
				),
			},
			{
				Config: testAccCheckCertificateWithStatusDataSourceConfig(domain, acm.CertificateStatusIssued),
				Check: resource.ComposeTestCheckFunc(
					//lintignore:AWSAT001
					resource.TestMatchResourceAttr(resourceName, "arn", arnRe),
					resource.TestCheckResourceAttr(resourceName, "status", acm.CertificateStatusIssued),
					resource.TestCheckResourceAttrSet(resourceName, "certificate"),
					resource.TestCheckResourceAttrSet(resourceName, "certificate_chain"),
				),
			},
			{
				Config: testAccCheckCertificateWithTypesDataSourceConfig(domain, acm.CertificateTypeAmazonIssued),
				Check: resource.ComposeTestCheckFunc(
					//lintignore:AWSAT001
					resource.TestMatchResourceAttr(resourceName, "arn", arnRe),
					resource.TestCheckResourceAttrSet(resourceName, "certificate"),
					resource.TestCheckResourceAttrSet(resourceName, "certificate_chain"),
				),
			},
			{
				Config: testAccCheckCertificateWithMostRecentDataSourceConfig(domain, true),
				Check: resource.ComposeTestCheckFunc(
					//lintignore:AWSAT001
					resource.TestMatchResourceAttr(resourceName, "arn", arnRe),
					resource.TestCheckResourceAttrSet(resourceName, "certificate"),
					resource.TestCheckResourceAttrSet(resourceName, "certificate_chain"),
				),
			},
			{
				Config: testAccCheckCertificateWithMostRecentAndStatusDataSourceConfig(domain, acm.CertificateStatusIssued, true),
				Check: resource.ComposeTestCheckFunc(
					//lintignore:AWSAT001
					resource.TestMatchResourceAttr(resourceName, "arn", arnRe),
					resource.TestCheckResourceAttrSet(resourceName, "certificate"),
					resource.TestCheckResourceAttrSet(resourceName, "certificate_chain"),
				),
			},
			{
				Config: testAccCheckCertificateWithMostRecentAndTypesDataSourceConfig(domain, acm.CertificateTypeAmazonIssued, true),
				Check: resource.ComposeTestCheckFunc(
					//lintignore:AWSAT001
					resource.TestMatchResourceAttr(resourceName, "arn", arnRe),
					resource.TestCheckResourceAttrSet(resourceName, "certificate"),
					resource.TestCheckResourceAttrSet(resourceName, "certificate_chain"),
				),
			},
		},
	})
}
```

Reach out if you have any questions! I'm also happy to help contribute to the work here :)

File: internal/service/acm/certificate_data_source.go [hunk 1]
```diff
@@ -25,6 +25,14 @@ func DataSourceCertificate() *schema.Resource {
 				Type:     schema.TypeString,
 				Computed: true,
 			},
+			"tls_certificate": {
```
[the remark this answers is not in the document]
Comment (anGie44) [reply internal/service/acm/certificate_data_source.go:28]: do you mind also adding the param to https://github.com/hashicorp/terraform-provider-aws/blob/main/website/docs/d/acm_certificate.html.markdown#attributes-reference ?

File: internal/service/acm/certificate_data_source.go [hunk 2]
```diff
@@ -25,6 +25,14 @@ func DataSourceCertificate() *schema.Resource {
...
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"tls_certificate_full_chain": {
```
[the remark this answers is not in the document]
Comment (anGie44) [reply internal/service/acm/certificate_data_source.go:32]: do you mind also adding the param to https://github.com/hashicorp/terraform-provider-aws/blob/main/website/docs/d/acm_certificate.html.markdown#attributes-reference ?

**Comment by rizkybiz** [comment]
@anGie44 I believe I've taken care of all of your concerns. I appreciate the great suggestions, (basically fixed it all yourself) and let me know if there's anything else that needs to be done to get these changes merged.

**Review by anGie44 — APPROVED** [review]
Changes looks great @rizkybiz ! Thanks so much 🚀 

Output of subset of acceptance tests:
```
--- PASS: TestAccACMCertificateDataSource_singleIssued (49.25s)
--- PASS: TestAccACMCertificateDataSource_noMatchReturnsError (12.72s)
```

**Pull request merged**